validate/error.go

```diff
@@ -0,0 +1,69 @@
+package validate
+
+import (
+	"errors"
+	"fmt"
+	"strings"
+)
+
+// ComplianceLevel represents the OCI compliance levels
+type ComplianceLevel int
+
+const (
+	ComplianceOptional ComplianceLevel = iota
+	ComplianceMay
+	ComplianceRecommended
+	ComplianceShould
+	ComplianceShouldNot
+	ComplianceShall
+	ComplianceShallNot
+	ComplianceRequired
+	ComplianceMustNot
+	ComplianceMust
+)
+
+// OCIErrorCode represents the compliance content
+type OCIErrorCode int
+
+const (
+	DefaultFilesystems OCIErrorCode = iota
+)
+
+// OCIError represents an error with compliance level and OCI reference
+type OCIError struct {
+	Level     ComplianceLevel
+	Reference string
+	Err       error
+}
+
+//FIXME: change to tagged spec releases
```

Comment:
How about add a field (not in this PR) like this :
       SpecVersions []string

So each OCI Error will be associate with one or more spec versions.
In runtimetest, we can add another option:
       spec-version.
So if an OCI Error does not have the wanted spec-version,  we can ignore it.